ts/src/index.ts

```diff
@@ -37,12 +37,16 @@ export const heap = {
 
 // If loaded with --require, start profiling.
 if (module.parent && module.parent.id === 'internal/preload') {
-  const stop = time.start();
-  process.on('exit', () => {
-    // The process is going to terminate imminently. All work here needs to
-    // be synchronous.
-    const profile = stop();
-    const buffer = encodeSync(profile);
-    writeFileSync(`pprof-profile-${process.pid}.pb.gz`, buffer);
+  time.start().then(stop => {
+    process.on('exit', async () => {
+      // The process is going to terminate imminently. All work here needs to
+      // be synchronous.
+
+      // TODO: this code no longer works because stop() cannot be run synchronously. Maybe
+      // beforeExit event would be a decent middleground, or can run this in a signal handler.
```

Comment:
Any thoughts on this?

Reply:
This is a breaking change, but the profiling agent for Node.js doesn't use this, so the breaking change seems fine to me.

I don't think it will impact integration tests ([busybench.ts](https://github.com/google/pprof-nodejs/blob/5ae0e9b2581cf62d051ef151925d0b1eb79e9d9c/system-test/busybench/src/busybench.ts) and [busybench.js](https://github.com/google/pprof-nodejs/blob/main/system-test/busybench-js/src/busybench.js) both invoke the profiler from their code, so should still work), so (beyond perhaps noting this change in release note), eliminating this should be too much effort.

Reply:
Made a follow up issue to remove the code and docs https://github.com/google/pprof-nodejs/issues/234